exec/java-exec/src/main/java/org/apache/drill/exec/expr/ExpressionTreeMaterializer.java

```diff
@@ -267,6 +267,17 @@ public LogicalExpression visitBooleanOperator(BooleanOperator op, FunctionLookup
       return new BooleanOperator(op.getName(), args, op.getPosition());
     }
 
+    static public int computePrecision(LogicalExpression currentArg) {
+        int precision = currentArg.getMajorType().getPrecision();
```

Comment:
Did you consider checking whether the type has a precision or not using getMajorType().hasPrecision()  ?  That way, you would only call getPrecision() if it returned True and otherwise set the precision for INT, BIGINT.